app/src/main/java/ca/klapstein/baudit/activities/PatientHomeActivity.java

```diff
@@ -90,11 +94,9 @@ public boolean onNavigationItemSelected(@NonNull MenuItem menuItem) {
         fab.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                Intent intent = new Intent(
-                        PatientHomeActivity.this,
-                        ProblemActivity.class
-                );
-                intent.putExtra("problemId", -1);
+                Intent intent = new Intent(getApplicationContext(), ProblemActivity.class);
+                intent.putExtra("problemPosition", -1);
+                intent.putExtra("mode", "edit");
```

Comment:
I think it would be best if these intent flags would be specified as public static final fields within their respective classes. Its safer, and cleaner.

Likely we will get code cleanliness bonus points.

Reply:
Agreed